WebService/v1/Controllers/DeviceModelsController.cs

```diff
@@ -12,22 +15,65 @@ namespace Microsoft.Azure.IoTSolutions.DeviceSimulation.WebService.v1.Controller
     public class DeviceModelsController : Controller
     {
         private readonly IDeviceModels deviceModelsService;
+        private readonly ILogger log;
 
-        public DeviceModelsController(IDeviceModels deviceModelsService)
+        public DeviceModelsController(
+            IDeviceModels deviceModelsService,
+            ILogger logger)
         {
             this.deviceModelsService = deviceModelsService;
+            this.log = logger;
         }
 
         [HttpGet]
-        public DeviceModelListApiModel Get()
+        public async Task<DeviceModelListApiModel> GetAsync()
         {
-            return DeviceModelListApiModel.FromServiceModel(this.deviceModelsService.GetList());
+            return DeviceModelListApiModel.FromServiceModel(await this.deviceModelsService.GetListAsync());
         }
 
         [HttpGet("{id}")]
-        public DeviceModelApiModel Get(string id)
+        public async Task<DeviceModelApiModel> GetAsync(string id)
         {
-            return DeviceModelApiModel.FromServiceModel(this.deviceModelsService.Get(id));
+            return DeviceModelApiModel.FromServiceModel(await this.deviceModelsService.GetAsync(id));
+        }
+
+        [HttpPost]
+        public async Task<DeviceModelApiModel> PostAsync(
+            [FromBody] DeviceModelApiModel deviceModel)
+        {
+            if (deviceModel == null)
+            {
+                this.log.Warn("No data provided", () => new { });
```

Comment:
nit: when logging without passing data, you can use `() => {}` instead of `() => new {}`